napari/layers/points/points.py

```diff
@@ -296,6 +300,7 @@ def __init__(
         property_choices=None,
         experimental_clipping_planes=None,
         shading='none',
+        experimental_canvas_size_limits=(0, 10000),
```

Comment:
Is the max size of `10000` significant? Can we do "np.inf"? or is that problematic for other reasons. I might do 2^16-1 or something really really big if we don't expect this to come into play.

Reply:
I don't know how `np.inf` would play here. I suppose we can/should give a max float value? Not sure what that is in glsl floats though. Certainly we can go bigger than 10000 though :P

Reply:
if `2^16-1` works then i think that is fine

Reply:
Oh, I just remembered why I simply put 10000: `gl_PointSize` is anyways limited by hardware/driver, and 10000 is way bigger than [anything we might even get close to](https://github.com/vispy/vispy/issues/2078#issuecomment-864078398). Don't forget these are screen pixels!

Reply:
oooh ok, that makes more sense then! let's leave at 10k then. i forgot they were screen pixels